lib/ansible/modules/network/itential/iap_token.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Itential <opensource@itential.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+This module provides the token for Itential Automation Platform
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: iap_token
+version_added: "2.8"
+author: "Itential (opensource@itential.com)"
+short_description: Get token for the Itential Automation Platform
+description:
+  - Checks the connection to IAP and retrieves a login token.
+options:
+  iap_port:
+    description:
+      - Provide the port number for the Itential Automation Platform
+    required: true
+    default: null
+
+  iap_fqdn:
+    description:
+      - Provide the fqdn for the Itential Automation Platform
```

Comment:
Assuming IP address can also work:
```suggestion
      - Provide the FQDN or IP address for the Itential Automation Platform
```
FQDN should be in caps